src/fn/jst.ts

```diff
@@ -4,58 +4,171 @@ import {
   type PcbSilkscreenPath,
 } from "circuit-json"
 import { z } from "zod"
-import { platedhole } from "src/helpers/platedhole"
-import { silkscreenRef, type SilkscreenRef } from "../helpers/silkscreenRef"
+
 import { platedHoleWithRectPad } from "src/helpers/platedHoleWithRectPad"
+import { rectpad } from "src/helpers/rectpad"
+import { silkscreenRef, type SilkscreenRef } from "../helpers/silkscreenRef"
 
 export const jst_def = z.object({
   fn: z.string(),
-  p: length.optional().default("2.2mm"),
-  id: length.optional().default("0.70mm"),
-  pw: length.optional().default("1.20mm"),
-  pl: length.optional().default("1.20mm"),
-  w: length.optional().default("6mm"),
-  h: length.optional().default("5mm"),
+  p: length.optional(),
+  id: length.optional(),
+  pw: length.optional(),
+  pl: length.optional(),
+  w: length.optional(),
+  h: length.optional(),
+  sh: z
```

Comment:
these need to have .describe!!!!!

particularly sh, and ph, but honestly ALL of them should have .describe!!!